Tribes/src/app/_components/soldiers/soldiers.component.ts

```diff
@@ -7,9 +11,53 @@ import { Component, OnInit } from '@angular/core';
 })
 export class SoldiersComponent implements OnInit {
 
-  constructor() { }
+  soldiers: Soldier[] = [];
+  levels: number[] = [];
+  level: number;
+  goldAmount: number;
+  costNewSoldier = COST_NEW_SOLDIER;
+
+  constructor(private soldiersService: SoldiersService, private resourceService: ResourceService) {
+    this.soldiersService.finishConstruction.subscribe({
+      next: () => {
+          this.getSoldiersByLevel();
+      }
+    });
+   }
 
   ngOnInit() {
+    this.getSoldiersByLevel();
+    this.createLevelArray();
+    this.getGoldAmount();
+
+  }
+
+  getSoldiersByLevel() {
+    this.soldiersService.getSoldiersFromAPI()
+    .subscribe(response => this.soldiers = response.soldiers);
+  }
+
```

Comment:
I may be wrong, but from what I can see, this returns the soldiers in total, not by level. Your next funciton filters by level.
If this is the case, I would recommend renaming the function to getSoldiers.